sdk/cosmos/azure_data_cosmos/src/clients/database_client.rs

```diff
@@ -74,7 +73,7 @@ impl DatabaseClientMethods for DatabaseClient {
         &self,
 
         #[allow(unused_variables)]
-        // This is a documented public API so prefixing with '_' is undesirable.
+        // REASON: This is a documented public API so prefixing with '_' is undesirable.
```

Comment:
Perhaps you're only seeing this lint because there's nothing public that actually uses it though, like an example. Should it be demonstrated instead? I'm not opposed to adding this, just wanted to point out that nothing public is touching it if you're seeing this.

Actually, that said, I thought as long as the trait is public, this shouldn't be raised. *So many* public APIs would trigger this lint otherwise. Maybe double check. Again, not opposed to the `#[allow]` attribute here.

Reply:
Y'know, it's probably that I put this here before adding the trait and then it became copy-pasta. I'll try removing it.

Reply:
I was able to get rid of all the others, but not this one... Honestly, I don't understand why this one needs it and the others don't. This method is used in examples, doctests, etc. just as much as the ones in `container_client`. I was able to actually remove those though. Only this one continues to report a warning.

Reply:
I didn't think it was use of the method, I thought it was use of the parameter within the method. In this case you never touch the options parameter, hence the lint warning.

Reply:
Correct. But that's true of the `options` parameter in the methods on `ContainerClient`. The `options` parameters aren't currently used at all, they're there just so we can add fields to them later.